File: leetcode/100-999/136. Single Number/solution_5.java
class Solution
{
    public int singleNumber(int[] nums)
    {
        HashMap<Integer, Integer> numsAndCount = new HashMap<>();

        for (int n : nums)
            numsAndCount.put(n, numsAndCount.getOrDefault(n, 0) + 1);

        for (var pair : numsAndCount.entrySet())
            if (pair.getValue() == 1)
                return pair.getKey();
        
        return -1;
    }
}

File: leetcode/100-999/136. Single Number/solution_5.cpp
class Solution 
{
    public:
        int singleNumber(vector<int>& nums) 
        {
            map<int, int> numsAndCount;

            for (int n : nums)
                numsAndCount[n]++;

            for (auto pair : numsAndCount)
                if (pair.second == 1)
                    return pair.first;

            return -1;
        }
};